src/large_pages/node_large_page.cc

```diff
@@ -138,18 +138,20 @@ static struct text_region FindNodeTextRegion() {
       std::string pathname;
       iss >> pathname;
       if (pathname == exename && permission == "r-xp") {
-        start = reinterpret_cast<uint64_t>(&__nodetext);
-        char* from = reinterpret_cast<char*>(hugepage_align_up(start));
-        char* to = reinterpret_cast<char*>(hugepage_align_down(end));
-
-        if (from < to) {
-          size_t size = to - from;
-          nregion.found_text_region = true;
-          nregion.from = from;
-          nregion.to = to;
-          nregion.total_hugepages = size / hps;
+        int64_t ntext = reinterpret_cast<uint64_t>(&__nodetext);
```

Comment:
I know you're following existing practices here but shouldn't that be `uintptr_t`?